solvexia_sdk/user/user.py

```diff
@@ -0,0 +1,42 @@
+import requests
+from solvexia_sdk import api
+
+class user:
+    def __init__(self, user_id):
+        self.user_id = user_id
+    
+    def get_user_list(self):
+        response = api.api_get("users")
+        api.status_code_check(response, "Error getting user list")
+        return response.json()
+    
+    def get_user(self):
+        response = api.api_get(f"users/{self.user_id}")
+        api.status_code_check(response, f"Error getting user with user_id {self.user_id}")
+        return response.json()
+
+    def disable_user(self):
+        payload = {
+            'accountActive': False
+        }
+        response = api.api_post(f"users/{self.user_id}", payload)
+        api.status_code_check(response, f"Error disabling user with user_id {self.user_id}")
+        return response.json()
+
+    def get_permissions(self):
+        response = api.api_get(f"users/{self.user_id}/permissions")
+        api.status_code_check(response, f"Error getting user's permissions with user_id {self.user_id}")
+        return response.json()
+
+    def update_permission(self, resource_id, role):
+        payload = {
+            'role': role
+        }
+        response = api.api_post(f"users/{self.user_id}/permissions/{resource_id}", payload)
+        api.status_code_check(response, f"Error updating user's permissions with user_id {self.user_id} for resource_id {resource_id}")
+        return response.json()
+    
+    def delete_permission(self, resource_id):
+        response = requests.delete(f"users/{self.user_id}/permissions/{resource_id}", headers = api.access_token)
```

Comment:
The url for requests.delete appears to be missing some parts.
i.e. "users/{self.user_id}/permissions/{resource_id}" is missing the "https://[...]/api/v1/" part that should come first.
So perhaps api.base_url + f"users/{self.user_id}/permissions/{resource_id}" would suffice.

This probably arose since the api module of the solvexia_sdk has api.api_get, api.api_post methods but does not have some kind of api.api_delete that wraps the requests call by default.

Reply:
@nullfrit Thanks for the detailed description. I have updated it. For delete_permission, I have remove the json return. It is now consistent with other delete methods.